{"level":"error","message":{"content":"Exception: Attendence ","meta":["TypeError: Cannot destructure property 'dayType' of 'req.dayType' as it is undefined.\n    at attendance (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\adminController.js:354:12)\n    at Layer.handle [as handle_request] (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at Immediate._onImmediate (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\multer\\lib\\make-middleware.js:53:37)\n    at process.processImmediate (node:internal/timers:478:21)"],"type":"USER"},"timestamp":"2023-09-14 10:12:49"}
{"level":"error","message":{"content":"Exception: Attendence ","meta":["ReferenceError: duration is not defined\n    at attendance (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\adminController.js:444:35)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"],"type":"USER"},"timestamp":"2023-09-14 11:24:33"}
{"level":"error","message":{"content":"attendance fetch failed ","meta":["ReferenceError: startDate is not defined\n    at getAttendance (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\userController.js:1329:24)\n    at Layer.handle [as handle_request] (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at isAuthenticate (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\common\\helpers\\auth.js:88:5)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"],"type":"USER"},"timestamp":"2023-09-14 12:05:06"}
{"level":"error","message":{"content":"attendance fetch failed ","meta":["ReferenceError: startDate is not defined\n    at getAttendance (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\userController.js:1329:24)\n    at Layer.handle [as handle_request] (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at isAuthenticate (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\common\\helpers\\auth.js:88:5)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"],"type":"USER"},"timestamp":"2023-09-14 12:05:06"}
{"level":"error","message":{"content":"attendance fetch failed ","meta":["CastError: Cast to date failed for value \"Invalid Date\" (type Date) at path \"date\" for model \"Attendance\"\n    at SchemaDate.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\date.js:376:11)\n    at SchemaDate.handleSingle (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\date.js:388:15)\n    at SchemaDate.castForQuery (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\date.js:419:18)\n    at cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\cast.js:334:39)\n    at Query.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4900:12)\n    at Query._castConditions (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2223:10)\n    at model.Query._find (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2249:8)\n    at model.Query.exec (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4419:28)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async getAttendance (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\userController.js:1328:28)"],"type":"USER"},"timestamp":"2023-09-14 12:05:37"}
{"level":"error","message":{"content":"attendance fetch failed ","meta":["CastError: Cast to date failed for value \"Invalid Date\" (type Date) at path \"date\" for model \"Attendance\"\n    at SchemaDate.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\date.js:376:11)\n    at SchemaDate.handleSingle (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\date.js:388:15)\n    at SchemaDate.castForQuery (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\date.js:419:18)\n    at cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\cast.js:334:39)\n    at Query.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4900:12)\n    at Query._castConditions (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2223:10)\n    at model.Query._find (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2249:8)\n    at model.Query.exec (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4419:28)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async getAttendance (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\userController.js:1328:28)"],"type":"USER"},"timestamp":"2023-09-14 12:05:39"}
{"level":"error","message":{"content":"attendance fetch failed ","meta":["CastError: Cast to date failed for value \"Invalid Date\" (type Date) at path \"date\" for model \"Attendance\"\n    at SchemaDate.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\date.js:376:11)\n    at SchemaDate.handleSingle (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\date.js:388:15)\n    at SchemaDate.castForQuery (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\date.js:419:18)\n    at cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\cast.js:334:39)\n    at Query.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4900:12)\n    at Query._castConditions (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2223:10)\n    at model.Query._find (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2249:8)\n    at model.Query.exec (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4419:28)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async getAttendance (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\userController.js:1328:28)"],"type":"USER"},"timestamp":"2023-09-14 12:06:27"}
{"level":"error","message":{"content":"attendance fetch failed ","meta":["CastError: Cast to date failed for value \"Invalid Date\" (type Date) at path \"date\" for model \"Attendance\"\n    at SchemaDate.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\date.js:376:11)\n    at SchemaDate.handleSingle (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\date.js:388:15)\n    at SchemaDate.castForQuery (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\date.js:419:18)\n    at cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\cast.js:334:39)\n    at Query.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4900:12)\n    at Query._castConditions (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2223:10)\n    at model.Query._find (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2249:8)\n    at model.Query.exec (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4419:28)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async getAttendance (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\userController.js:1328:28)"],"type":"USER"},"timestamp":"2023-09-14 12:06:27"}
{"level":"error","message":{"content":"Exception: Failed to update user personal details ","meta":["MongoServerError: Plan executor error during findAndModify :: caused by :: E11000 duplicate key error collection: cubedots-hrms.personaldetails index: personal_email_1 dup key: { personal_email: \"aniket@gmail.com\" }\n    at Connection.onMessage (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\connection.js:202:26)\n    at MessageStream.<anonymous> (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\connection.js:61:60)\n    at MessageStream.emit (node:events:513:28)\n    at processIncomingData (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\message_stream.js:124:16)\n    at MessageStream._write (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\message_stream.js:33:9)\n    at writeOrBuffer (node:internal/streams/writable:392:12)\n    at _write (node:internal/streams/writable:333:10)\n    at Writable.write (node:internal/streams/writable:337:10)\n    at TLSSocket.ondata (node:internal/streams/readable:766:22)\n    at TLSSocket.emit (node:events:513:28)\n    at addChunk (node:internal/streams/readable:324:12)\n    at readableAddChunk (node:internal/streams/readable:297:9)\n    at Readable.push (node:internal/streams/readable:234:10)\n    at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23)"],"type":"USER"},"timestamp":"2023-09-14 16:17:04"}
{"level":"error","message":{"content":"Exception: Failed to update user personal details ","meta":["MongoServerError: Plan executor error during findAndModify :: caused by :: E11000 duplicate key error collection: cubedots-hrms.personaldetails index: personal_email_1 dup key: { personal_email: \"aniket@gmail.com\" }\n    at Connection.onMessage (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\connection.js:202:26)\n    at MessageStream.<anonymous> (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\connection.js:61:60)\n    at MessageStream.emit (node:events:513:28)\n    at processIncomingData (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\message_stream.js:124:16)\n    at MessageStream._write (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\message_stream.js:33:9)\n    at writeOrBuffer (node:internal/streams/writable:392:12)\n    at _write (node:internal/streams/writable:333:10)\n    at Writable.write (node:internal/streams/writable:337:10)\n    at TLSSocket.ondata (node:internal/streams/readable:766:22)\n    at TLSSocket.emit (node:events:513:28)\n    at addChunk (node:internal/streams/readable:324:12)\n    at readableAddChunk (node:internal/streams/readable:297:9)\n    at Readable.push (node:internal/streams/readable:234:10)\n    at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23)"],"type":"USER"},"timestamp":"2023-09-14 16:54:42"}
{"level":"error","message":{"content":"Exception: Failed to update user personal details ","meta":["MongoServerError: Plan executor error during findAndModify :: caused by :: E11000 duplicate key error collection: cubedots-hrms.personaldetails index: personal_email_1 dup key: { personal_email: \"aniket@gmail.com\" }\n    at Connection.onMessage (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\connection.js:202:26)\n    at MessageStream.<anonymous> (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\connection.js:61:60)\n    at MessageStream.emit (node:events:513:28)\n    at processIncomingData (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\message_stream.js:124:16)\n    at MessageStream._write (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\message_stream.js:33:9)\n    at writeOrBuffer (node:internal/streams/writable:392:12)\n    at _write (node:internal/streams/writable:333:10)\n    at Writable.write (node:internal/streams/writable:337:10)\n    at TLSSocket.ondata (node:internal/streams/readable:766:22)\n    at TLSSocket.emit (node:events:513:28)\n    at addChunk (node:internal/streams/readable:324:12)\n    at readableAddChunk (node:internal/streams/readable:297:9)\n    at Readable.push (node:internal/streams/readable:234:10)\n    at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23)"],"type":"USER"},"timestamp":"2023-09-14 16:54:49"}
{"level":"error","message":{"content":"Exception: Failed to update user personal details ","meta":["MongoServerError: Plan executor error during findAndModify :: caused by :: E11000 duplicate key error collection: cubedots-hrms.personaldetails index: personal_email_1 dup key: { personal_email: \"aniket@gmail.com\" }\n    at Connection.onMessage (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\connection.js:202:26)\n    at MessageStream.<anonymous> (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\connection.js:61:60)\n    at MessageStream.emit (node:events:513:28)\n    at processIncomingData (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\message_stream.js:124:16)\n    at MessageStream._write (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongodb\\lib\\cmap\\message_stream.js:33:9)\n    at writeOrBuffer (node:internal/streams/writable:392:12)\n    at _write (node:internal/streams/writable:333:10)\n    at Writable.write (node:internal/streams/writable:337:10)\n    at TLSSocket.ondata (node:internal/streams/readable:766:22)\n    at TLSSocket.emit (node:events:513:28)\n    at addChunk (node:internal/streams/readable:324:12)\n    at readableAddChunk (node:internal/streams/readable:297:9)\n    at Readable.push (node:internal/streams/readable:234:10)\n    at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23)"],"type":"USER"},"timestamp":"2023-09-14 16:56:15"}
{"level":"error","message":{"content":"Exception: Failed to update user personal details ","meta":["CastError: Cast to ObjectId failed for value \"{ _id: 6 }\" (type Object) at path \"_id\" for model \"Role\"\n    at ObjectId.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\objectid.js:250:11)\n    at SchemaType.applySetters (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1220:12)\n    at SchemaType.castForQuery (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1632:15)\n    at cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\cast.js:298:34)\n    at Query.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4900:12)\n    at Query._castConditions (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2223:10)\n    at model.Query._findOne (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2510:8)\n    at model.Query.exec (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4419:28)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async isAuthorized (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\common\\helpers\\auth.js:94:25)\n    at async updatePersonalDetails (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\employeeController.js:80:24)"],"type":"USER"},"timestamp":"2023-09-14 17:28:46"}
{"level":"error","message":{"content":"Exception: Failed to update user personal details ","meta":["CastError: Cast to ObjectId failed for value \"{ _id: 6 }\" (type Object) at path \"_id\" for model \"Role\"\n    at ObjectId.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\objectid.js:250:11)\n    at SchemaType.applySetters (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1220:12)\n    at SchemaType.castForQuery (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1632:15)\n    at cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\cast.js:298:34)\n    at Query.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4900:12)\n    at Query._castConditions (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2223:10)\n    at model.Query._findOne (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2510:8)\n    at model.Query.exec (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4419:28)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async isAuthorized (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\common\\helpers\\auth.js:94:25)\n    at async updatePersonalDetails (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\employeeController.js:80:24)"],"type":"USER"},"timestamp":"2023-09-14 17:28:53"}
{"level":"error","message":{"content":"Exception: Failed to update user personal details ","meta":["CastError: Cast to ObjectId failed for value \"{ _id: 6 }\" (type Object) at path \"_id\" for model \"Role\"\n    at ObjectId.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\objectid.js:250:11)\n    at SchemaType.applySetters (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1220:12)\n    at SchemaType.castForQuery (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1632:15)\n    at cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\cast.js:298:34)\n    at Query.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4900:12)\n    at Query._castConditions (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2223:10)\n    at model.Query._findOne (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2510:8)\n    at model.Query.exec (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4419:28)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async isAuthorized (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\common\\helpers\\auth.js:94:25)\n    at async updatePersonalDetails (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\employeeController.js:80:24)"],"type":"USER"},"timestamp":"2023-09-14 17:29:44"}
{"level":"error","message":{"content":"Exception: Failed to update user personal details ","meta":["CastError: Cast to ObjectId failed for value \"{ _id: 6 }\" (type Object) at path \"_id\" for model \"Role\"\n    at ObjectId.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\objectid.js:250:11)\n    at SchemaType.applySetters (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1220:12)\n    at SchemaType.castForQuery (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1632:15)\n    at cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\cast.js:298:34)\n    at Query.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4900:12)\n    at Query._castConditions (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2223:10)\n    at model.Query._findOne (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2510:8)\n    at model.Query.exec (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4419:28)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async isAuthorized (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\common\\helpers\\auth.js:94:25)\n    at async updatePersonalDetails (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\employeeController.js:80:24)"],"type":"USER"},"timestamp":"2023-09-14 17:30:15"}
{"level":"error","message":{"content":"Exception: Failed to update user personal details ","meta":["CastError: Cast to ObjectId failed for value \"{ _id: 6 }\" (type Object) at path \"_id\" for model \"Role\"\n    at ObjectId.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\objectid.js:250:11)\n    at SchemaType.applySetters (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1220:12)\n    at SchemaType.castForQuery (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1632:15)\n    at cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\cast.js:298:34)\n    at Query.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4900:12)\n    at Query._castConditions (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2223:10)\n    at model.Query._findOne (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2510:8)\n    at model.Query.exec (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4419:28)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async isAuthorized (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\common\\helpers\\auth.js:94:25)\n    at async updatePersonalDetails (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\employeeController.js:79:24)"],"type":"USER"},"timestamp":"2023-09-14 17:30:35"}
{"level":"error","message":{"content":"Exception: Failed to update user personal details ","meta":["CastError: Cast to ObjectId failed for value \"{ _id: 6 }\" (type Object) at path \"_id\" for model \"Role\"\n    at ObjectId.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\objectid.js:250:11)\n    at SchemaType.applySetters (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1220:12)\n    at SchemaType.castForQuery (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1632:15)\n    at cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\cast.js:298:34)\n    at Query.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4900:12)\n    at Query._castConditions (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2223:10)\n    at model.Query._findOne (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2510:8)\n    at model.Query.exec (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4419:28)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async isAuthorized (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\common\\helpers\\auth.js:94:25)\n    at async updatePersonalDetails (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\employeeController.js:79:24)"],"type":"USER"},"timestamp":"2023-09-14 17:31:28"}
{"level":"error","message":{"content":"Exception: Failed to update user personal details ","meta":["CastError: Cast to ObjectId failed for value \"{ _id: 6 }\" (type Object) at path \"_id\" for model \"Role\"\n    at ObjectId.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\objectid.js:250:11)\n    at SchemaType.applySetters (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1220:12)\n    at SchemaType.castForQuery (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1632:15)\n    at cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\cast.js:298:34)\n    at Query.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4900:12)\n    at Query._castConditions (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2223:10)\n    at model.Query._findOne (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2510:8)\n    at model.Query.exec (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4419:28)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async isAuthorized (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\common\\helpers\\auth.js:94:25)\n    at async updatePersonalDetails (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\employeeController.js:79:24)"],"type":"USER"},"timestamp":"2023-09-14 17:32:02"}
{"level":"error","message":{"content":"Exception: Failed to update user personal details ","meta":["ReferenceError: roeId is not defined\n    at isAuthorized (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\common\\helpers\\auth.js:94:17)\n    at updatePersonalDetails (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\employeeController.js:79:30)\n    at Layer.handle [as handle_request] (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at isAuthenticate (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\common\\helpers\\auth.js:88:5)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"],"type":"USER"},"timestamp":"2023-09-14 17:33:25"}
{"level":"error","message":{"content":"Exception: Failed to update user personal details ","meta":["CastError: Cast to ObjectId failed for value \"{ _id: 6 }\" (type Object) at path \"_id\" for model \"Role\"\n    at ObjectId.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\objectid.js:250:11)\n    at SchemaType.applySetters (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1220:12)\n    at SchemaType.castForQuery (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1632:15)\n    at cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\cast.js:298:34)\n    at Query.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4900:12)\n    at Query._castConditions (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2223:10)\n    at model.Query._findOne (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2510:8)\n    at model.Query.exec (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4419:28)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async isAuthorized (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\common\\helpers\\auth.js:95:25)\n    at async updatePersonalDetails (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\employeeController.js:79:24)"],"type":"USER"},"timestamp":"2023-09-14 17:33:51"}
{"level":"error","message":{"content":"Exception: Failed to update user personal details ","meta":["CastError: Cast to ObjectId failed for value \"{ _id: 6 }\" (type Object) at path \"_id\" for model \"Role\"\n    at ObjectId.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schema\\objectid.js:250:11)\n    at SchemaType.applySetters (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1220:12)\n    at SchemaType.castForQuery (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\schematype.js:1632:15)\n    at cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\cast.js:298:34)\n    at Query.cast (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4900:12)\n    at Query._castConditions (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2223:10)\n    at model.Query._findOne (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:2510:8)\n    at model.Query.exec (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\query.js:4419:28)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async isAuthorized (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\common\\helpers\\auth.js:95:25)\n    at async updatePersonalDetails (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\employeeController.js:79:24)"],"type":"USER"},"timestamp":"2023-09-14 17:34:18"}
{"level":"error","message":{"content":"Exception: Failed to update user family details ","meta":["ReferenceError: adminRole is not defined\n    at updateFamilyDetails (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\employeeController.js:169:43)\n    at Layer.handle [as handle_request] (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at isAuthenticate (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\common\\helpers\\auth.js:88:5)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"],"type":"USER"},"timestamp":"2023-09-14 17:50:48"}
{"level":"error","message":{"content":"Exception: Failed to update user expeience details ","meta":["ReferenceError: adminRole is not defined\n    at updateExperienceDetails (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\controllers\\employeeController.js:261:43)\n    at Layer.handle [as handle_request] (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\express\\lib\\router\\route.js:144:13)\n    at isAuthenticate (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\src\\common\\helpers\\auth.js:88:5)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"],"type":"USER"},"timestamp":"2023-09-14 17:56:29"}
{"level":"error","message":{"content":"User cannot login: ","meta":["MongooseError: Operation `users.findOne()` buffering timed out after 10000ms\n    at Timeout.<anonymous> (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\drivers\\node-mongodb-native\\collection.js:185:23)\n    at listOnTimeout (node:internal/timers:569:17)\n    at process.processTimers (node:internal/timers:512:7)"],"type":"USER"},"timestamp":"2023-09-14 18:22:45"}
{"level":"error","message":{"content":"User cannot login: ","meta":["MongooseError: Operation `users.findOne()` buffering timed out after 10000ms\n    at Timeout.<anonymous> (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\drivers\\node-mongodb-native\\collection.js:185:23)\n    at listOnTimeout (node:internal/timers:569:17)\n    at process.processTimers (node:internal/timers:512:7)"],"type":"USER"},"timestamp":"2023-09-14 18:23:16"}
{"level":"error","message":{"content":"User cannot login: ","meta":["MongooseError: Operation `users.findOne()` buffering timed out after 10000ms\n    at Timeout.<anonymous> (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\drivers\\node-mongodb-native\\collection.js:185:23)\n    at listOnTimeout (node:internal/timers:569:17)\n    at process.processTimers (node:internal/timers:512:7)"],"type":"USER"},"timestamp":"2023-09-14 18:23:58"}
{"level":"error","message":{"content":"User cannot login: ","meta":["MongooseError: Operation `users.findOne()` buffering timed out after 10000ms\n    at Timeout.<anonymous> (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\drivers\\node-mongodb-native\\collection.js:185:23)\n    at listOnTimeout (node:internal/timers:569:17)\n    at process.processTimers (node:internal/timers:512:7)"],"type":"USER"},"timestamp":"2023-09-14 18:24:17"}
{"level":"error","message":{"content":"User cannot login: ","meta":["MongooseError: Operation `users.findOne()` buffering timed out after 10000ms\n    at Timeout.<anonymous> (E:\\Cubedots-HRMS\\cubedots-hrms-backend\\node_modules\\mongoose\\lib\\drivers\\node-mongodb-native\\collection.js:185:23)\n    at listOnTimeout (node:internal/timers:569:17)\n    at process.processTimers (node:internal/timers:512:7)"],"type":"USER"},"timestamp":"2023-09-14 18:24:31"}
